module/VuFind/src/VuFind/View/Helper/Root/DeferredRecommend.php

```diff
@@ -0,0 +1,69 @@
+<?php
+
+/**
+ * Deferred recommendation module view helper
+ *
+ * PHP version 8
+ *
+ * Copyright (C) Villanova University 2023.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 2,
+ * as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ *
+ * @category VuFind
+ * @package  View_Helpers
+ * @author   Demian Katz <demian.katz@villanova.edu>
+ * @author   Maccabee Levine <msl321@lehigh.edu>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org/wiki/development Wiki
+ */
+
+namespace VuFind\View\Helper\Root;
+
+use VuFind\Recommend\RecommendInterface;
+
+/**
+ * Deferred recommendation module view helper
+ *
+ * @category VuFind
+ * @package  View_Helpers
+ * @author   Demian Katz <demian.katz@villanova.edu>
+ * @author   Maccabee Levine <msl321@lehigh.edu>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org/wiki/development Wiki
+ */
+class DeferredRecommend extends \Laminas\View\Helper\AbstractHelper
+{
+    use ClassBasedTemplateRendererTrait;
+
+    /**
+     * Render code to load a recommendation module via AJAX.
+     *
+     * @param RecommendInterface $recommend The recommendation object to render
+     * @param string             $location  Recommendation location (side, top)
+     * @param int                $index     Index of the recommendation configuration
+     *
+     * @return string
+     */
+    public function __invoke(
+        $urlParams,
+        $containerId,
+    ) {
+        $template = 'Recommend/Deferred.phtml';
+        $className = $this::class;
+        $context = [
+            'containerId' => $containerId,
+        ];
+        return $this->renderClassTemplate($template, $className, $context);
```

Comment:
I don't think I need a "class based" template here, except maybe for caching benefit.  But I didn't see a clear alternative that appeared to work in this context.

Reply:
If you don't need the class-based functionality, can you just do `$this->getView()->render('Recommend/Deferred.phtml', $context)`?

Reply:
Thanks, I didn't think of getView().  But as it turns out I do appear to need some of that functionality, otherwise `$this->recommend` is undefined in the template, and if I have to pass that in the context somehow, it seems simpler to just use the class template method?

Reply:
I guess you need the context helper if you want to render in the current context... so you could either leave it as is (which is doing some extra work), or use this code:

```
        $view = $this->getView();
        $contextHelper = $view->plugin('context');
        $oldContext = $contextHelper($view)->apply($context);
        $html = $view->render('Recommend/Deferred.phtml');
        $contextHelper($view)->restore($oldContext);
        return $html;
```

Reply:
Or if you don't want to bother with the context helper, and you only need to worry about container ID, you might be able to do it like this:

```
        $view = $this->getView();
        $oldContainerId = $view->containerId ?? null;
        $view->containerId = $containerId;
        $html = $view->render('Recommend/Deferred.phtml');
        $view->containerId = $oldContainerId;
        return $html;
```

Reply:
I used the context helper for flexibility in case there are some other params we need to pass later.